src/components/Laatija/Laatijat.svelte

```diff
@@ -1,9 +1,307 @@
+  import Input from '@Component/Input/Input';
+  import Button from '@Component/Button/Button';
+  import * as laatijaApi from '@Component/Laatija/laatija-api';
+  import * as geoApi from '@Component/Geo/geo-api';
+  import { locale, _ } from '@Language/i18n';
+  import * as locales from '@Language/locale-utils';
+  import * as LaatijaUtils from './laatija-utils';
+
+  import * as Future from '@Utility/future-utils';
+
+  import { flashMessageStore } from '@/stores';
+
+  let laatijat = Maybe.None();
+  let itemsPerPage = 20;
+
+  const fields = [
+    { id: 'laatija', title: $_('laatija.laatija') },
+    { id: 'henkilotunnus', title: $_('laatija.henkilotunnus') },
+    {
+      id: 'patevyystaso',
+      title: $_('laatija.patevyystaso')
+    },
+    {
+      id: 'toteamispaivamaara',
+      title: $_('laatijahaku.voimassaolo'),
+      format: LaatijaUtils.formatVoimassaoloaika
+    },
+    { id: 'toimintaalue', title: $_('laatija.paatoimintaalue') },
+    { id: 'postinumero', title: $_('laatija.postinumero') },
+    {
+      id: 'yritys',
+      type: 'action-with-template',
+      title: $_('yritys.yritykset'),
+      actionTemplate: ({ id, nimi }) => ({
+        type: 'link',
+        href: `#/yritys/${id}`,
+        text: `${nimi}`
+      })
+    }
+  ];
+
+  const formatLocale = R.curry((localizations, id) =>
+    R.compose(
+      locales.label($locale),
+      R.find(R.propEq('id', id))
+    )(localizations)
+  );
+
+  const findYritysById = R.curry((yritykset, id) =>
+    R.find(R.propEq('id', id), yritykset)
+  );
+
+  const formatYritys = R.curry((yritykset, idt) =>
+    R.compose(
+      R.map(R.pick(['id', 'nimi'])),
+      R.map(findYritysById(yritykset))
+    )(idt)
+  );
+
+  const formatLaatija = R.curry((patevyydet, yritykset, toimintaalueet) =>
+    R.compose(
+      R.map(laatija =>
+        R.compose(
+          R.pick(
+            R.compose(
+              R.append('id'),
+              R.append('voimassa'),
+              R.map(R.prop('id'))
+            )(fields)
+          ),
+          R.assoc(
+            'patevyystaso',
+            formatLocale(patevyydet, laatija.patevyystaso)
+          ),
+          R.assoc(
+            'toimintaalue',
+            formatLocale(toimintaalueet, laatija.toimintaalue)
+          ),
+          R.assoc('yritys', formatYritys(yritykset, laatija.yritys)),
+          R.assoc('laatija', `${laatija.etunimi} ${laatija.sukunimi}`)
+        )(laatija)
+      )
+    )
+  );
+
+  const getLaatijat = () =>
+    R.compose(
+      Future.fork(
+        R.compose(
+          flashMessageStore.add('Laatija', 'error'),
+          R.always($_('errors.load-error'))
+        ),
+        R.compose(([result, yritykset, patevyydet, toimintaalueet]) => {
+          laatijat = Maybe.Some(
+            formatLaatija(patevyydet, yritykset, toimintaalueet)(result)
+          );
+        })
+      ),
+      Future.parallel(5),
+      R.append(geoApi.toimintaalueet),
+      R.append(laatijaApi.patevyydet),
```

Comment:
Nää itseasiassa kaipaa refaktorointia. Tällähetkellä utilsissa nähtävästi on ulkoisena riippuvuutena fetch, joka luetaan globaalista nimiavaruudesta. Ne pitäisi laittaa muotoon missä niille ensimmäisenä parametrinä dependency injectataan tuo fetch. Näin pysyvät puhtaina ja testattavina. Myöskin tuon parallel-promisen rakentaminen helpottuu kun ei tarvitse erikseen appendata näitä vaan ne voi laittaa tuohon alempaan juxtiin mukaan.